### DISCUSSION TITLE
We could be using Leaflet.js and it's vector features to visualize the J4 workflows. Why not?

### DISCUSSION THREAD

**Discussion by alexandreelise** [joomla/joomla-cms#35238] [2. Cold Features]
Hello Super Joomlers,

The open source javascript high performance and low memory and file size footprint called Leaflet.js could be used to render the workflow nodes even on a huge map due to its zoom , pan etc, features just like a real map. The thing is instead of rendering gps coordinates as points etc. We might render svg or canvas shapes on the map with cartesian coordinates or custom coordinate system. Here is the link to the project: [Get Leaflet](https://leafletjs.com/)     . Here is the story behind Leaflet: [Leaflet story](https://www.youtube.com/watch?v=NLbyHffKQuU)

Share your thoughts about this. Do you think it's worth a try?

**Comment by alexandreelise** [comment]
Hi Benjamin @bembelimen what do you think about this?

**Reply by bembelimen** [reply]
We're looking for a graphical solution for a longer time now, especially something which allows live editing of the workflow and does not break with a lot of transitions. So if this is possible why not, someone has just to implement it :)

**Reply by alexandreelise** [reply]
Here is a starting point we could look into. [Non Geographic Maps](https://leafletjs.com/examples/crs-simple/crs-simple.html)

**Reply by dgrammatiko** [reply]
Meh, I wouldn't use this library for workflows, there are other helpers that fit way better the given task. My 2c

**Reply by dgrammatiko** [reply]
Eg: 
demo: https://alyssax.com/x/flowy
repo: https://github.com/alyssaxuu/flowy

**Reply by alexandreelise** [reply]
Hi Dimitris. The app you showed is nice but can it handle huge workflows? Leaflet is insanely fast but maybe I am wrong and it's not the right tool for this use case.

**Reply by dgrammatiko** [reply]
Define **huge**. 
Also (unless the code was fundamentally changed, I haven't followed the improvements in that space) a reasonable amount of workflow steps will deem Joomla unusable due to poorly designed SQL queries
